appdevguide/IOCErrorLogging.md

```diff
@@ -0,0 +1,306 @@
+
+Normally the errlog system displays all messages on the console.
+`eltc` can be used to suppress these messages.
+
+- {external+epics-base:cpp:func}`eltc`
+- {external+epics-base:cpp:func}`errlogInit`
+- {external+epics-base:cpp:func}`errlogInit2`
+
+eltc determines if errlog task writes message to the console.
+During error message storms this command can be used to suppress console messages.
+A argument of 0 suppresses the messages, any other value lets messages go to the console.
+
+`errlogInit` or `errlogInit2` can be used to initialize the error logging system 
+with a larger buffer and maximum message size.
+
+## Status codes
+
+EPICS defined status values provide the following features:
+
+- Whenever possible, IOC routines return a status value: 0 means OK, non-0 means an error.
+- The header files for most IOC subsystems contain macros defining error status symbols and strings.
+- Routines are provided for run time access of the error status symbols and strings.
+- A global variable `errVerbose` helps code decide if error messages should be generated.
+
+IOC routines often return a [long integer](https://en.wikipedia.org/wiki/Integer_(computer_science)#Long_integer)
+status value, encoded similar to vxWorks error status encoding.
+On some modern architectures a long integer is more than 32 bits wide, but in order to keep the API compatible the status values are still passed as long integers, even though only 32 bits are ever used.
+The most significant 16 bits indicate the subsystem or module where the error occurred.
+The least significant 16 bits contain a subsystem-specific status value.
+In order that status values do not conflict with the vxWorks error status values, all subsystem numbers are greater than 500.
+
+A header file `errMdef.h` defines macros for all the subsystem numbers.
+For example the database access routines use this module number:
+
+`#define M_dbAccess  (511 << 16)   /*Database Access Routines*/`
+
+There are header files for every IOC subsystem that returns standard status values.
+The status values are encoded with lines of the following format:
+
+`#define S_xxxxxxx value /*string value*/`
+
+For example:
+
+`#define S_dbLib_recNotFound (M_dbLib|5)        /* Record Not Found */`
+
+For example, when {external+epics-base:cpp:func}`dbFindRecordPart` cannot find a record, it executes the statement:
+
+`return S_dbLib_recNotFound;`
+
+The calling routine checks the return status as follows:
+
+```
+status = dbGetField(...);
+if(status) {/* Call was not successful */ }
+```
+(ioclog-chapter)=
+## iocLog
+
+IOC logging comprises two modules:
+`iocLogServer` and `iocLogClient`.
+
+See {external+epics-base::doc}`logClient_h` for up-to-date API information.
+
+The client code runs on each IOC and listens for the messages generated locally by the [errlog](#err-log-overview) system.
+It also reports the messages from the vxWorks logMsg facility.
+
+### iocLogServer
```

Comment:
While this section probably needs to be included here for now, I would like to see the iocLogServer marked as deprecated, advising new EPICS sites to look at other standard logging servers such as Logstash which can provide much better user interfaces.

Reply:
I agree. This felt quite outdated but I did not want to drop it just yet. I can make a note that this is deprecated and maybe write a few lines about possible alternatives.